samtranslator/model/eventsources/pull.py

```diff
@@ -48,6 +49,10 @@ def to_cloudformation(self, **kwargs):
         except NotImplementedError:
             function_name_or_arn = function.get_runtime_attr("arn")
 
+        if not self.Stream and not self.Queue:
```

Comment:
Can you instead add a validate functino to each subclass that gets called here? Each subclass can then validate and return a customized error message. What do you think?

Reply:
I deliberately went with this simpler approach since this is temporary code and I think the error message is clear enough even when they are combined for both cases.